### ISSUE TITLE
Add WebSocket resource endpoint 

### ISSUE THREAD

**Issue by gryphon2411** [sphinx-contrib/httpdomain#25]
As an HTTP API docs writer,
I want to have web-socket resource .rst directives,
So I can doc a web-socket resource endpoint, just like I doc HTTP resource endpoint.

As of today, doc of web-socket API resource endpoints is not out-of-the-box of this extension.

For HTTP APIs, there is a possibility to have web-socket resource endpoints (additional to the HTTP resource endpoints).

Suggested directives syntax:
```
.. http:ws:: <url>

:server-event <event-name>:
:client-event <event-name>:
```

Example, for a web-socket API endpoint (in the light of the this extension official docs):
```
.. http:ws:: /users/(int:user_id)/posts/

   Returns a post that the user (`user_id`) just added.

   :reqheader cookie:
   :server-event add:

   **Example response**:

   .. sourcecode:: json

        {
          "post_id": 54321,
          "author_id": 123,
          "tags": ["server", "web"],
          "subject": "Tomcat doesn't work"
        }

   :statuscode 401: user is not authorized.
```
Might be rendered to:
![rendered web-socket directive](https://user-images.githubusercontent.com/11075654/59195435-a3245a00-8b94-11e9-9352-d60c8630e4d7.png)

**Comment by gryphon2411** [comment]
@stevepiercy Thanks

I revisited this feature request and would be happy to work on it.

I think a small separate `websocket` Sphinx domain may be clearer than `http:ws`, since a WebSocket connection is not an HTTP method. The initial scope I have in mind is deliberately small: a `websocket:connection` directive, directional `server-message` / `client-message` lists, and a separate WebSocket routing table.

For example:

```rst
.. websocket:connection:: /chat/(str:room)

   :server-message message: Broadcasts a chat message to room members.
   :client-message message: Sends a chat message to the room.
```

This would leave handshake details, message schemas, close codes, and framework discovery as ordinary prose or future work.

Does that direction sound reasonable before I start a PR?

**Comment by stevepiercy** [comment]
@gryphon2411 yes, please go ahead! Two other people liked your suggestion, @loelkes and @ilan-theodoro, so perhaps they would also like to collaborate, review, or offer suggestions.
